Source/Extensions/Result.swift

```diff
@@ -4,9 +4,10 @@ import Result
 public extension Result {
   static func fromDecoded<T>(decoded: Decoded<T>) -> Result<T, NSError> {
     switch decoded {
-    case let .Success(obj): return .Success(obj)
-    case .TypeMismatch, .MissingKey:
-      return .Failure(error(decoded.description))
+    case let .Success(obj):
+      return .Success(obj)
+    case let .Failure(decodedError):
+      return .Failure(error(decodedError.description))
```

Comment:
since `decodeError` is an `ErrorType`, it can just be cast to `NSError`:

```swift
return .Failure(decodedError as NSError)
```

_but_ we can actually just improve the overall type here and make this return a `Result<T, DecodeError>`. I only used `NSError` because there _wasn't_ a concrete `ErrorType` to use initially.